src/panels/config/automation/ha-automation-editor.ts

```diff
@@ -944,7 +947,31 @@ export class HaAutomationEditor extends PreventUnsavedMixin(
 
         // wait for automation to appear in entity registry when creating a new automation
         if (entityRegPromise) {
+          const timeout = setTimeout(() => {
+            this._readOnly = true;
+            this._dirty = false;
+            showAlertDialog(this, {
+              title: this.hass.localize(
+                "ui.panel.config.automation.editor.new_automation_setup_failed_title",
+                {
+                  type: this.hass.localize(
+                    "ui.panel.config.automation.editor.type_automation"
+                  ),
+                }
+              ),
+              text: this.hass.localize(
+                "ui.panel.config.automation.editor.new_automation_setup_failed_text",
+                {
+                  type: this.hass.localize(
+                    "ui.panel.config.automation.editor.type_automation"
+                  ),
+                }
+              ),
+              warning: true,
+            });
+          }, 5000);
           const automation = await entityRegPromise;
```

Comment:
Can we instead use the `promiseTimeout` helper?
```suggestion
	  try {
          	const automation = await promiseTimeout(5000, entityRegPromise);
           } catch (e) {
           	showAlertDialog();
           }
```

Reply:
I almost used this, but I didn't discover clean way to detect if the error caught was from the timeout or from the entityRegPromise. If it's not a timeout, I want to fall through to the regular error handling that already existed. 

I could catch the error and see if the string regex contained the words "Timed out" but seemed a bit messy.

Reply:
Agree, we should make the reject an instanceof [Error](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error) instead of just a string, that we can then give custom name:

```ts
class TimeoutError extends Error {
  public timeout: number;

  constructor(timeout: number, ...params) {
    super(...params);

    // Maintains proper stack trace for where our error was thrown (only available on V8)
    if (Error.captureStackTrace) {
      Error.captureStackTrace(this, TimeoutError);
    }

    this.name = "TimeoutError";
    // Custom debugging information
    this.timeout = timeout;
    this.message = `Timed out in ${timeout} ms.`;
  }
}

export const promiseTimeout = (ms: number, promise: Promise<any> | any) => {
  const timeout = new Promise((_resolve, reject) => {
    setTimeout(() => {
      reject(new TimeoutError(ms));
    }, ms);
  });

  // Returns a race between our timeout and the passed in promise
  return Promise.race([promise, timeout]);
};
```